### PR TITLE
cabal-update-shell-hook update cabal if it is bad in shellHook

### PR DESCRIPTION
**Pull request by yaitskov**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -99,7 +99,12 @@ in {
     exactDeps = false;
   }; in {
     inherit hsPkgs;
-    ormoluShell = shellFor (ps: [ ps.ormolu ]);
+    ormoluShell = (shellFor (ps: [ ps.ormolu ]))
```
Comment (mrkkrp) [inline default.nix:102]: It bother me to modify the state of the system implicitly like this, but I will leave the decision to @amesgen.

File: default.nix [hunk 2]
```diff
@@ -99,7 +99,12 @@ in {
...
+      .overrideAttrs(oa: { shellHook = (oa.shellHook or "") + ''
+                             cabal exec -v0 -- ghc --print-libdir && echo "Cabal is good" || \
+                               { echo "Update Cabal"; cabal update; }
+                           '';
+                         });
```
Comment (amesgen) [inline default.nix:107]:
```suggestion
    ormoluShell = shellFor (ps: [ ps.ormolu ps.extract-hackage-info ]);
```

**Review by amesgen — CHANGES_REQUESTED** [review]
Thanks; I think a conceptually somewhat nicer way to fix this is to have switch to having just one combined shell for `ormolu` and `extract-hackage-info`, as mentioned in https://github.com/tweag/ormolu/issues/973#issuecomment-1387092995 and already done in https://github.com/tweag/ormolu/pull/940, as it avoids touching the global system state, as mentioned by @mrkkrp.

For this, feel free to just apply this diff: https://gist.github.com/amesgen/11eaf0ee65796f97059873016d027bf4

File: default.nix (hunk 1 above)
Comment (amesgen) [reply default.nix:102]: I suggested an alternative that does not have this caveat: https://github.com/tweag/ormolu/pull/974#pullrequestreview-1262270371

**Comment by amesgen** [comment]
#940 has been merged, which also resolved #973 as per https://github.com/tweag/ormolu/issues/973#issuecomment-1387092995.

**Pull request closed without merging**